### PR TITLE
MON-3900: Remove prometheus-adapter code

### PR DESCRIPTION
**Pull request by slashpai**
pkg: remove prometheus-adapter code

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: pkg/manifests/types.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ type AlertmanagerMainConfig struct {
 }
 
 // The `K8sPrometheusAdapter` resource defines settings for the Prometheus Adapter component.
-// This is deprecated and will be removed in a future version.
+// This is deprecated config, setting this has no effect and will be removed in a future version.
```
Comment (slashpai) [inline pkg/manifests/types.go:107]: Keeping this to check in 4.18 release if any user defined config exists in cmo's configmap and then block upgrades till removed

**Comment by slashpai** [comment]
cc @simonpasquier @jan--f @machine424

File: pkg/manifests/types.go (hunk 1 above)
Comment (simonpasquier) [reply pkg/manifests/types.go:107]: Can you add a TODO comment? It doesn't have to be 4.18 "exactly" but setting a target is good.

**Comment by slashpai** [comment]
@simonpasquier could you please review again?

**Comment by jan--f** [comment]
Looks good! Happy to merge when CI and Simon are happy :)

**Comment by slashpai** [comment]
I will rebase once https://github.com/openshift/cluster-monitoring-operator/pull/2414 is merged to fix generate issue

**Comment by madeelrh** [comment]
Why was it removed? and what it is being replaced with? It is not mentioned in this PR.

**Comment by slashpai** [comment]
This PR was part of clean up. Its been replaced with metrics-server for resource metrics API 
You can find the details in https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

**Pull request merged**